test/00_transform/MatMul.cu

```diff
@@ -101,136 +105,156 @@ TYPED_TEST(MatMulTestFloatTypes, SmallRect)
 {
   MATX_ENTER_HANDLER();
   using TestType = cuda::std::tuple_element_t<0, TypeParam>;
-  constexpr index_t m = 4;
-  constexpr index_t k = 8;
-  constexpr index_t n = 16;
-  tensor_t<TestType, 2> a{{m, k}};
-  tensor_t<TestType, 2> b{{k, n}};
-  tensor_t<TestType, 2> c{{m, n}};
-
-  this->pb->template InitAndRunTVGenerator<TestType>(
-      "00_transforms", "matmul_operators", "run", {m, k, n});
-
-  this->pb->NumpyToTensorView(a, "a");
-  this->pb->NumpyToTensorView(b, "b");
-
-  // example-begin matmul-test-1
-  // Perform the GEMM C = A*B
-  (c = matmul(a, b)).run();
-  // example-end matmul-test-1
-  MATX_TEST_ASSERT_COMPARE(this->pb, c, "c", this->thresh);
-
+  using ExecType = cuda::std::tuple_element_t<1, TypeParam>;
+  if constexpr (!detail::CheckMatMulSupport<ExecType, TestType>()) {
```

Comment:
Is this needed every time if it's done in `SetUp`?

Reply:
Without the check in the test itself, it will fail a static assert in matmul_impl when it tries to compile the test with a half precision datatype.